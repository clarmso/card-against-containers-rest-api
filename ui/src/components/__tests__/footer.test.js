import React from "react"
import renderer from "react-test-renderer"
import Footer from "../footer"

describe("<Footer />", () => {
  test("Display Footer with default parameters", () => {
    const component = renderer.create(<Footer />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("Display Footer with author, homepage link and github link", () => {
    const author = "Hello World"
    const homepageLink = "https://helloworld.com"
    const githubLink = "https://github.com/helloworld"
    const component = renderer.create(
      <Footer
        author={author}
        homepageLink={homepageLink}
        githubLink={githubLink}
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
