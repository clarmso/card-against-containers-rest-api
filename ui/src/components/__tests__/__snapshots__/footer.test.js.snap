// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Display Footer with author, homepage link and github link 1`] = `
<div
  className="MuiGrid-root"
>
  <br />
  <footer
    className="makeStyles-footer-1"
  >
    <p>
      Made with
       
      <span
        aria-label="heart"
        role="img"
      >
        <span
          className="makeStyles-srOnly-2"
        >
          Love
        </span>
        ❤️
         
      </span>
       
      by 
      <a
        href="https://helloworld.com"
      >
        Hello World
      </a>
       © 
      2021
      .
       
      <a
        href="https://github.com/helloworld"
      >
        <span
          className="makeStyles-srOnly-2"
        >
          Github Repo
        </span>
        <svg
          aria-hidden={true}
          aria-label="github"
          className="MuiSvgIcon-root makeStyles-icon-3"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
          />
        </svg>
      </a>
    </p>
  </footer>
</div>
`;

exports[`<Footer /> Display Footer with default parameters 1`] = `
<div
  className="MuiGrid-root"
>
  <br />
  <footer
    className="makeStyles-footer-1"
  >
    <p>
      Made with
       
      <span
        aria-label="heart"
        role="img"
      >
        <span
          className="makeStyles-srOnly-2"
        >
          Love
        </span>
        ❤️
         
      </span>
       
      by 
      <a
        href=""
      >
        
      </a>
       © 
      2021
      .
       
      <a
        href="https://github.com/clarmso/cards-against-containers"
      >
        <span
          className="makeStyles-srOnly-2"
        >
          Github Repo
        </span>
        <svg
          aria-hidden={true}
          aria-label="github"
          className="MuiSvgIcon-root makeStyles-icon-3"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
          />
        </svg>
      </a>
    </p>
  </footer>
</div>
`;
